tower-http/src/services/fs/mod.rs

```diff
@@ -213,3 +214,76 @@ fn response_from_io_error(
         _ => Err(err),
     }
 }
+
+struct LastModified(HttpDate);
+
+impl From<SystemTime> for LastModified {
+    fn from(time: SystemTime) -> Self {
+        LastModified(time.into())
+    }
+}
+
+struct IfUnmodifiedSince(HttpDate);
+struct IfModifiedSince(HttpDate);
+
+impl IfModifiedSince {
+    /// Check if the supplied time means the resource has been modified.
+    fn is_modified(&self, last_modified: &LastModified) -> bool {
+        self.0 < last_modified.0
+    }
+
+    /// convert a header value into a IfModifiedSince, invalid values are silentely ignored
+    fn from_header_value(value: &http::header::HeaderValue) -> Option<IfModifiedSince> {
+        String::from_utf8(value.as_bytes().to_vec())
+            .ok()
+            .map(|value| httpdate::parse_http_date(&value).ok())
```

Comment:
`map` followed by `flatten` should be `flat_map`

Reply:
yes! you're right. (btw this is an option, as there is no `flat_map` on that type, I always forget I can use `and_then` to do that.)